src/editor/TextEditor.java

```diff
@@ -1,11 +1,252 @@
+
+        save.addActionListener( actionEvent ->
+                FileLoader.saveFile( fileName, textArea.getText(), true));
+
+        saveAs.addActionListener( actionEvent ->
+                FileLoader.saveFile( null, textArea.getText(), true));
+
+        exit.addActionListener( actionEvent ->
+                {
+                    FileLoader.saveFile( fileName, textArea.getText(), true);
+                    dispose();
+                }
+        );
+
+        file.add(load);
+        file.add(save);
+        file.add(saveAs);
+        file.addSeparator();
+        file.add(exit);
+
+        return file;
+    }
+
+    private JMenu menuSearchCreator(){
+        JMenu search = new JMenu("Search");
+
+        JMenuItem start = new JMenuItem("Start search");
+        JMenuItem prev = new JMenuItem("Previous match");
+        JMenuItem next = new JMenuItem("Next match");
+        JMenuItem useRegularex = new JMenuItem("Use regular expressions");
+
+
+        useRegularex.addActionListener( actionEvent -> {
+                useRegex.setSelected(!useRegex.isSelected());
+                textSearch.search();
+            }
+        );
+        start.addActionListener( actionEvent -> textSearch.search());
+        next.addActionListener( actionEvent -> textSearch.nextMatch());
+        prev.addActionListener( actionEvent -> textSearch.prevMatch());
+
+        search.add(start);
+        search.add(prev);
+        search.add(next);
+        search.add(useRegularex);
+
+        return search;
+    }
+
+    private JPanel createFileSelectGroup(){
+        ImageIcon saveIcon = new ImageIcon("img/save.png");
+        ImageIcon openIcon = new ImageIcon("img/open.png");
+        ImageIcon findIcon = new ImageIcon("img/find.png");
+        ImageIcon leftIcon = new ImageIcon("img/left.png");
+        ImageIcon rightIcon = new ImageIcon("img/right.png");
+
+        JButton load = new JButton(openIcon);
+        JButton save = new JButton(saveIcon);
+        JButton find = new JButton(findIcon);
+        JButton left = new JButton(leftIcon);
+        JButton right = new JButton(rightIcon);
+
+        load.setPreferredSize(new Dimension(40,40));
+        save.setPreferredSize(new Dimension(40,40));
+        find.setPreferredSize(new Dimension(40,40));
+        left.setPreferredSize(new Dimension(40,40));
+        right.setPreferredSize(new Dimension(40,40));
+        useRegex.setFont(font);
+
+        load.setBorderPainted(false);
+        save.setBorderPainted(false);
+        find.setBorderPainted(false);
+        left.setBorderPainted(false);
+        right.setBorderPainted(false);
+
+        load.setBackground(backColor);
+        save.setBackground(backColor);
+        find.setBackground(backColor);
+        left.setBackground(backColor);
+        right.setBackground(backColor);
+        useRegex.setBackground(backColor);
+
+        load.addActionListener( actionEvent -> FileChooser.open());
```

Comment:
why an extra space here? Try to use load.addActionListener(actionEvent -> FileChooser.open());